cli/main.c

```diff
@@ -1497,8 +1497,8 @@ static int fw_read(int argc, char **argv)
 	}
 
 	progress_start();
-	ret = switchtec_fw_read_fd(cfg.dev, cfg.out_fd, inf->part_addr,
-				   inf->image_len, progress_update);
+	ret = switchtec_fw_img_write_body(cfg.dev, cfg.out_fd,
```

Comment:
Ick, I very much prefer a name that reflects the fact that it's reading it from the device.